include/zephyr/drivers/fuel_gauge.h

```diff
@@ -255,6 +264,7 @@ typedef int (*fuel_gauge_battery_cutoff_t)(const struct device *dev);
 
 __subsystem struct fuel_gauge_driver_api {
 	fuel_gauge_get_property_t get_property;
+	fuel_gauge_get_properties_t get_properties;
```

Comment:
I don't think this is a good idea, you'd end up driver implementing only one of the two and it'll all become a big mess, just have `get_properties` outside of the api struct and leave only the subsystem implementation. Let's clean up the drivers with a funny props implementation separately.

Reply:
> you'd end up driver implementing only one of the two and it'll all become a big mess,

Is there something wrong with a driver only implementing one of the two? IMO, that's not terribly messy. How about I hide this behind a common config like: `CONFIG_FUEL_GAUGE_CUSTOM_GET_PROPS_FN` ?

Reply:
yeah, one could implement the multi but not the single and then applications using the single would not work with it, makes no sense, but also like keeping it simple, there's no need for the dual api

Reply:
As discussed offline: Adding comment to explicitly call out the multi-get/set API field as a possible future direction but leaving it out until we have a concrete upstream use-case.